src/canon.rs

```diff
@@ -136,33 +136,38 @@ pub fn canon_path<T: Into<String>>(inpath: T) -> String {
 mod tests {
     use super::*;
 
+    // Assert that canon path equals expected path with different path separators
+    fn assert_canon_path_eq(left: &[&str], right: &[&str]) {
+        for path_sep in ["/", "\\"] {
+            assert_eq!(canon_path(left.join(path_sep)), right.join(path_sep));
+        }
+    }
+
     #[test]
     fn noop() {
-        assert_eq!(canon_path("foo"), "foo");
-
-        assert_eq!(canon_path("foo/bar"), "foo/bar");
+        assert_canon_path_eq(&["foo"], &["foo"]);
+        assert_canon_path_eq(&["foo", "bar"], &["foo", "bar"]);
     }
 
     #[test]
     fn dot() {
-        assert_eq!(canon_path("./foo"), "foo");
-        assert_eq!(canon_path("foo/."), "foo/");
-        assert_eq!(canon_path("foo/./bar"), "foo/bar");
+        assert_canon_path_eq(&[".", "foo"], &["foo"]);
+        assert_canon_path_eq(&["foo", "."], &["foo", ""]);
+        assert_canon_path_eq(&["foo", ".", "bar"], &["foo", "bar"]);
     }
 
     #[test]
     fn slash() {
-        assert_eq!(canon_path("/foo"), "/foo");
-        assert_eq!(canon_path("foo//bar"), "foo/bar");
+        assert_canon_path_eq(&["", "foo"], &["", "foo"]);
```

Comment:
I think these would be better if assert_canon_path_eq accepted strings with "/" and internally did the / => \ replacement.